packages/zod-validator/src/index.ts

```diff
@@ -3,8 +3,8 @@ import { validator } from 'hono/validator'
 import type { z, ZodSchema, ZodError } from 'zod'
 
 export type Hook<T, E extends Env, P extends string, O = {}> = (
```

Comment:
We could the target as another generic here: 

```ts
export type Hook<T, E extends Env, P extends string, Target extends keyof ValidationTargets = keyof ValidationTargets, O = {}> = (
```

we can then pass the `Target` generic of `zValidator` to the hook

```ts
  hook?: Hook<z.infer<T>, E, P, Target>
```

and have the exact target reflected in the hook:

<img width="977" alt="Bildschirmfoto 2024-09-24 um 11 37 29" src="https://github.com/user-attachments/assets/55515aa4-a567-49a3-855d-a4807a596424">